iot_packet_verifier/src/burner.rs

```diff
@@ -58,6 +61,42 @@ where
         }
     }
 
+    pub async fn recover_attempted_burns(&mut self) -> Result<(), BurnError<P::Error, S::Error>> {
+        tracing::info!("Attempting to recover attempted burns");
+
+        for attempted_burn in self
+            .pending_burns
+            .fetch_incomplete_burns()
+            .await
+            .map_err(BurnError::SqlError)?
+        {
+            let amount = attempted_burn.amount();
+            let since = attempted_burn.latest_transaction_signature()?;
+            tracing::info!(%since, %amount, "Checking chain for burn transaction");
+            if self
+                .solana
+                .has_burn_transaction(&attempted_burn.payer, amount, &since)
+                .await
+                .map_err(BurnError::SolanaError)?
+            {
+                tracing::info!("Found a matching transaction. Removing burn attempt");
+                self.pending_burns
+                    .complete_burn_attempt(&attempted_burn.payer, amount)
```

Comment:
On the regular burn function, the payer balance is updated after a complete_burn_attempt is successful. Does the same need performed here after this run of complete_burn_attempt?